src/rules/unbound-method.ts

```diff
@@ -79,6 +79,16 @@ export default createRule<Options, MessageIds>({
       ...baseSelectors,
       MemberExpression(node: TSESTree.MemberExpression): void {
         if (node.parent?.type === AST_NODE_TYPES.CallExpression) {
+          if (
+            node.parent.callee.type === AST_NODE_TYPES.MemberExpression &&
+            node.parent.callee.object.type === AST_NODE_TYPES.Identifier &&
+            node.parent.callee.object.name === 'jest' &&
+            node.parent.callee.property.type === AST_NODE_TYPES.Identifier &&
+            node.parent.callee.property.name === 'mocked'
+          ) {
+            return;
+          }
```

Comment:
you leverage `parseJestFnCall` for this - you can see an example of how to use the result for this in `no-confusing-set-timeout`:

```
function isJestSetTimeout(jestFnCall: ParsedJestFnCall) {
  return (
    jestFnCall.type === 'jest' &&
    jestFnCall.members.length === 1 &&
    isIdentifier(jestFnCall.members[0], 'setTimeout')
  );
}
```